### PR TITLE
Document process for closing trial accounts

### PR DESCRIPTION
**Pull request by kr8n3r**
What
----

Move the process to close trial accounts from the [doc](https://docs.google.com/document/d/1ialNFhD4S38ZK3R2skgwCEMQJHQCRqnbODNF-GXmyCw/edit#) into a webpage

How to review
-------------

- clone branch
- run `bundle exec middleman server`
- visit lives /accounts/suspending_old_trial_accounts

Who can review
--------------
not @kr8n3r 

**Link to the page**
Screenshot 2020-10-29 at 09 44 34

*The page*
gds6352 local_4567_accounts_closing_trial_accounts_

### CODE REVIEW

File: source/accounts/closing_trial_accounts.html.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Process for closing GOV.UK PaaS trial accounts
+
+This process describes how we close old trial accounts to retain historical billing data and prevent their use ongoing and the ability to run up infrastructure bills.
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:3]:
```suggestion
This process describes how we close old trial accounts to retain historical billing data and prevent running up infrastructure bills.
```

File: source/accounts/closing_trial_accounts.html.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+|2|Trial|A tenant with a 90 day trial account with the ability to deploy apps and provision small backing services|90 days|
+|3|Billable|A full tenant with a billing arrangement that is being recharged for the resource consumed|Indefinitely|
+|4|Suspended|The org is suspended and the tenant will not be able to perform any operations on apps or services.<br /><br />Users will be able to use the apps. Tenants will be able to log in to CloudFoundry.|30 days|
+|5|Halted|The apps are halted/stopped and users will receive 404 error messages from the platform.<br /><br />The data is preserved and the account will still run up bills for the databases|30 days|
+|6|Removed|The application instances and services are deleted by an operator.<br /><br />Users will see no difference compared to a halted/stopped app, but the tenant will have to re-push the app and re-create any backing services to restore their apps.<br /><br />Export final bill and any necessary data.<br /><br />At this point the account cannot incur any billable events and the bill goes to zero.<br /><br />All apps and services and data unavailable.|Indefinitely|
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:24]:
```suggestion
|6|Removed|The application instances and services are deleted by an operator.<br /><br />Users will see no difference compared to a halted/stopped app, but the tenant will have to re-push the app and re-create any backing services to restore their apps.<br /><br />At this point the account cannot incur any billable events and the bill goes to zero.<br /><br />All apps and services and data unavailable.|Indefinitely|
```

File: source/accounts/closing_trial_accounts.html.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Accounts closure process
+
+1. Review trial organisations<br />
+  - [in London region](https://admin.london.cloud.service.gov.uk/reports/organisations)
+  - [in Ireland region](https://admin.cloud.service.gov.uk/reports/organisations)
+  - Look for accounts that are 2.5 months in to their 3 month trial period, and still on the trial quota
+1. Update [backlog list of potential accounts](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=527069954) for closure (in PaaS adoption tracker)
+1. Keep track of [potential accounts for closure](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=949444725) in tracker while contacting org managers 
+1. Keep track of [closed accounts and cost savings](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=159027804) by adding each trial account to the tracker
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:37]: Thanks, all of these link to the correct spreadsheet

File: source/accounts/closing_trial_accounts.html.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+1. If tenant confirms they intend to keep using their account, they need to [switch to billable](https://team-manual.cloud.service.gov.uk/accounts/account_lifecycle/#upgrading-trial-orgs-to-paid-orgs) - support engineer and DM to help (track via zendesk)
+If tenant says they don’t want the account, go to state 6 (Removed)
+1. If no response from tenant after 2 weeks grace period the support engineer suspends the account - State 4 (Suspended)
+1. Notify the tenant that their trial account has been suspended and they have 30 days to tell us they need the account. Make it clear that after this timeframe, we will stop their apps running, but keep their data State 5 (Halted)
+1. If tenant responds, they need to [switch to billable](https://team-manual.cloud.service.gov.uk/accounts/account_lifecycle/#upgrading-trial-orgs-to-paid-orgs) - support engineer and DM to help (track via Zendesk)
+If tenant says they don’t want the account, go to state 6 (Removed) 
+1. If no response from tenant, after the 1 month grace period the support engineer stops apps running state 5 (Halted)
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:45]:
```suggestion
1. If no response from tenant, after the 30 day grace period the support engineer stops apps running state 5 (Halted)
```

File: source/accounts/closing_trial_accounts.html.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+1. Notify the tenant via Zendesk that we have stopped their apps services
+1. Pull billing data associated with the trial account and save?
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:47]:
```suggestion
1. Pull billing data associated with the trial account and save in [PaaS team folder] (https://drive.google.com/drive/folders/1aq0gSOrDS1EGsq0aEx4dIOYEmy9j5dJL)
```

File: source/accounts/closing_trial_accounts.html.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+1. Notify tenant via Zendesk that in 30 days we will delete any data and all of their apps
+1. If no response from tenant, after the 1 month grace period the support engineer deletes all apps and services, and removes users from the org state 6 (removed)
```
Comment (Jenny3020) [inline source/accounts/closing_trial_accounts.html.md:49]:
```suggestion
1. If no response from tenant, after the 30 day grace period the support engineer deletes all apps and services, and removes users - org state 6 (removed)
```

**Review by AP-Hunt — APPROVED** [review]
This looks and reads like I'd expect

**Review by Jenny3020 — COMMENTED** [review]
I've reviewed as someone who will run through this process and help close the trial accounts. Thanks Jani!

**Comment by kr8n3r** [comment]
@Jenny3020 I have accepted all the suggestions

File: source/accounts/closing_trial_accounts.html.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
+# Process for closing GOV.UK PaaS trial accounts
+
+This process describes how we close old trial accounts to retain historical billing data and prevent running up infrastructure bills.
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:3]:
```suggestion
This process describes how we close old trial accounts to retain historical billing data and prevent infrastructure bills running up.
```

File: source/accounts/closing_trial_accounts.html.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+A closed account:
+
+  - is unusable by the tenant (no `cf push`, `cf start`. `cf stop`, `cf scale`)
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:7]:
```suggestion
  - is unusable by the tenant (no `cf push`, `cf start`, `cf stop` or `cf scale`)
```

File: source/accounts/closing_trial_accounts.html.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  - has no services or data 
+  - has no applications 
+  - is flagged as suspended
+  - has no users associated with the Cloud Foundry org
+  - incurs no costs to us
+  - exists as an empty shell in Cloud Foundry
+
+![A state diagram showing the possible routes for tenants adopting the paas](/diagrams/process-for-closing-old-trial-accounts.svg)
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:15]: Broken link

File: source/accounts/closing_trial_accounts.html.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+| Sequence | Title | Description | Max time |
+|:---|:---|:---|:---|
+|1|New tenant|A new tenant that does not have a paas account yet|Indefinitely|
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:19]:
```suggestion
|1|New tenant|A new tenant that does not have a PaaS account yet|Indefinitely|
```

File: source/accounts/closing_trial_accounts.html.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+|2|Trial|A tenant with a 90 day trial account with the ability to deploy apps and provision small backing services|90 days|
+|3|Billable|A full tenant with a billing arrangement that is being recharged for the resource consumed|Indefinitely|
+|4|Suspended|The org is suspended and the tenant will not be able to perform any operations on apps or services.<br /><br />Users will be able to use the apps. Tenants will be able to log in to CloudFoundry.|30 days|
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:22]:
```suggestion
|4|Suspended|The org is suspended and the tenant will not be able to perform any operations on apps or services.<br /><br />Users will be able to use the apps. Tenants will be able to log in to Cloud Foundry.|30 days|
```

File: source/accounts/closing_trial_accounts.html.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+|5|Halted|The apps are halted/stopped and users will receive 404 error messages from the platform.<br /><br />The data is preserved and the account will still run up bills for the databases|30 days|
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:23]:
```suggestion
|5|Halted|The apps are halted/stopped and users will receive '404' error messages from the platform.<br /><br />The data is preserved and the account will still run up bills for the databases|30 days|
```

Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:23]: Is there a difference in functionality between 'halted' and 'stopped'?

File: source/accounts/closing_trial_accounts.html.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+|6|Removed|The application instances and services are deleted by an operator.<br /><br />Users will see no difference compared to a halted/stopped app, but the tenant will have to re-push the app and re-create any backing services to restore their apps.<br /><br />At this point the account cannot incur any billable events and the bill goes to zero.<br /><br />All apps and services and data unavailable.|Indefinitely|
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:24]:
```suggestion
|6|Removed|The application instances and services are deleted by an operator.<br /><br />Users will see no difference compared to a halted/stopped app, but the tenant will have to re-push the app and re-create any backing services to restore their apps.<br /><br />At this point the account cannot incur any billable events and the bill goes to zero.<br /><br />All app, services and data are unavailable.|Indefinitely|
```

File: source/accounts/closing_trial_accounts.html.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+![BAU trial account expiry process](/diagrams/BAU-trial-account-expiry-process-map.svg)
+
+
+## Accounts closure process
+
+1. Review trial organisations<br />
+  - [in London region](https://admin.london.cloud.service.gov.uk/reports/organisations)
+  - [in Ireland region](https://admin.cloud.service.gov.uk/reports/organisations)
+  - Look for accounts that are 2.5 months in to their 3 month trial period, and still on the trial quota
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:34]: This shouldn't be a bullet point

File: source/accounts/closing_trial_accounts.html.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+|5|Halted|The apps are halted/stopped and users will receive 404 error messages from the platform.<br /><br />The data is preserved and the account will still run up bills for the databases|30 days|
+|6|Removed|The application instances and services are deleted by an operator.<br /><br />Users will see no difference compared to a halted/stopped app, but the tenant will have to re-push the app and re-create any backing services to restore their apps.<br /><br />At this point the account cannot incur any billable events and the bill goes to zero.<br /><br />All apps and services and data unavailable.|Indefinitely|
+
+![BAU trial account expiry process](/diagrams/BAU-trial-account-expiry-process-map.svg)
+
+
+## Accounts closure process
+
+1. Review trial organisations<br />
+  - [in London region](https://admin.london.cloud.service.gov.uk/reports/organisations)
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:32]:
```suggestion
  - [in the London region](https://admin.london.cloud.service.gov.uk/reports/organisations)
```

File: source/accounts/closing_trial_accounts.html.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+  - [in Ireland region](https://admin.cloud.service.gov.uk/reports/organisations)
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:33]:
```suggestion
  - [in the Ireland region](https://admin.cloud.service.gov.uk/reports/organisations)
```

File: source/accounts/closing_trial_accounts.html.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+  - Look for accounts that are 2.5 months in to their 3 month trial period, and still on the trial quota
+1. Update [backlog list of potential accounts](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=527069954) for closure (in PaaS adoption tracker)
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:35]:
```suggestion
1. Update the [backlog list of potential accounts](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=527069954) for closure (in PaaS adoption tracker)
```

File: source/accounts/closing_trial_accounts.html.md (hunk 14 above)
Comment (kr8n3r) [reply source/accounts/closing_trial_accounts.html.md:34]: tech docs markdown parse can't handle breaks in lists unfortunately

File: source/accounts/closing_trial_accounts.html.md (hunk 17 above)
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md]: Points in numbered lists should have full stops, according to the style guide

File: source/accounts/closing_trial_accounts.html.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+1. Keep track of [potential accounts for closure](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=949444725) in tracker while contacting org managers
```
Comment (cmcnallygds) [inline source/accounts/closing_trial_accounts.html.md:36]:
```suggestion
1. Keep track of [potential accounts for closure](https://docs.google.com/spreadsheets/d/1bZP7W-5nJxDicJ2lc_eT873zXXjXnHh-qGYJobHUIv8/edit#gid=949444725) in the tracker while contacting org managers 
```

File: source/accounts/closing_trial_accounts.html.md (hunk 12 above)
Comment (kr8n3r) [reply source/accounts/closing_trial_accounts.html.md]: maybe @AP-Hunt or @pauldougan knows?

Comment (AP-Hunt) [reply source/accounts/closing_trial_accounts.html.md]: The account state is halted, and the apps are stopped. We don't use the word halted with apps.

Comment (kr8n3r) [reply source/accounts/closing_trial_accounts.html.md]: update the sentence to "The apps are stopped and users will receive..." then ?

Comment (kr8n3r) [reply source/accounts/closing_trial_accounts.html.md]: sentence updated

**Comment by kr8n3r** [comment]
all @cmcnallygds changes accepted

**Pull request merged**